src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1337,12 +1337,12 @@ private static Task ExecuteTask<T>(Task<T> task, HttpContext httpContext)
 
         static async Task ExecuteAwaited(Task<T> task, HttpContext httpContext)
         {
-            await httpContext.Response.WriteAsJsonAsync(await task);
+            await httpContext.Response.WriteAsJsonAsync<object?>(await task);
```

Comment:
Could you add a comment about this? It would be trivially easy for someone to remove the `<object?>` because it feels uneccessary.. Alternatively, consider using the overload that accepts the type so it's way less likely to be refactored:

```suggestion
 
            var value = await task;
            // We want to always use the actual return type rather than the declared return type, hence we use value.GetType()
            await httpContext.Response.WriteAsJsonAsync(value, value.GetType()); 
```

Reply:
I added the following comment:

```C#
// Call WriteAsJsonAsync<object?>() to serialize the runtime return type rather than the declared return type.
```

I'm continuing to call `WriteAsJsonAsync<object?>(value)` because the alternative would be more like `WriteAsJsonAsync(value, value?.GetType() ?? typeof(object))`.